zebra-scan/src/storage/db/sapling.rs

```diff
@@ -141,6 +141,54 @@ impl Storage {
         keys
     }
 
+    /// Returns all the keys and their last scanned heights.
+    pub fn sapling_keys_and_last_scanned_heights(&self) -> HashMap<SaplingScanningKey, Height> {
+        let sapling_tx_ids = self.sapling_tx_ids_cf();
+        let mut keys = HashMap::new();
+
+        let last_stored_record: Option<(SaplingScannedDatabaseIndex, SaplingScannedResult)> =
```

Comment:
This is the other half of the serialization type fix in:
https://github.com/ZcashFoundation/zebra/pull/8080/files#r1421817246

```suggestion
        let last_stored_record: Option<(SaplingScannedDatabaseIndex, Option<SaplingScannedResult>)> =
```

Reply:
Done as part of https://github.com/ZcashFoundation/zebra/pull/8080/commits/ae5bc94e54f94fd9cb49e95ead4cf62184edaadb